src/extension.ts

```diff
@@ -18,21 +18,29 @@ export function activate(context: vscode.ExtensionContext) {
         vscode.window.showInputBox({
             placeHolder: 'Enter the name of the new terminal'
         }).then(name => {
+            for (let j = 0; j <_terminals.length; j++) {
+                _terminals[j].hide();
+            }
+            
             _terminals.push(new StatusBarTerminal(_terminalCounter++, name));
         });
     }));
 
     for (let i = 1; i <= MAX_TERMINALS; i++) {
         context.subscriptions.push(vscode.commands.registerCommand(`terminalTabs.showTerminal${i}`, (a) => {
-            _terminals[i - 1].show();
+            for (let j = 0; j <_terminals.length; j++) {
```

Comment:
_terminals.forEach would probably be better for readability/consistency as performance is no issue in this extension.